File: question1.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jan 16 13:57:45 2025

@author: arijh
"""

# Fonction pour calculer la somme des valeurs d'étalonnage dans un fichier
def extraire_somme_etalonnage(chemin_fichier):
    # Initialiser la somme des valeurs d'étalonnage
    somme_etalonnage = 0

    # Ouvrir le fichier en mode lecture
    with open(chemin_fichier, 'r') as fichier:
        # Initialiser le compteur de ligne
        index = 1
        # Parcourir chaque ligne du fichier
        for ligne in fichier:
            # Supprimer les espaces ou sauts de ligne inutiles
            ligne = ligne.strip()
            premier_chiffre = ''  # Variable pour stocker le premier chiffre trouvé
            dernier_chiffre = ''  # Variable pour stocker le dernier chiffre trouvé

            # Parcourir chaque caractère de la ligne pour trouver le premier chiffre
            for char in ligne:
                if char.isdigit():
                    premier_chiffre = char
                    break  # Arrêter dès que le premier chiffre est trouvé

            # Parcourir la ligne à l'envers pour trouver le dernier chiffre
            for char in reversed(ligne):
                if char.isdigit():
                    dernier_chiffre = char
                    break  # Arrêter dès que le dernier chiffre est trouvé

            # Si au moins un chiffre est trouvé, former le nombre à partir des chiffres
            if premier_chiffre or dernier_chiffre:
                nombre_forme = int(premier_chiffre + dernier_chiffre)
                # Afficher la concaténation des chiffres trouvés pour chaque ligne
                print(f"Ligne {index}: {premier_chiffre}{dernier_chiffre}")
                # Ajouter le nombre formé à la somme totale
                somme_etalonnage += nombre_forme
            else:
                print(f"Ligne {index}: Aucun chiffre trouvé")  # Afficher si aucun chiffre n'est trouvé

            # Incrémenter le compteur de ligne
            index += 1

    # Afficher la somme totale des valeurs d'étalonnage
    print(f"\nLa somme des valeurs d'étalonnage est : {somme_etalonnage}")

# Définir le chemin du fichier contenant les données
chemin_fichier = "C:/Users/arijh/Desktop/Test_Arij_Hamraoui/document.txt"

# Appeler la fonction pour calculer et afficher la somme des valeurs d'étalonnage
extraire_somme_etalonnage(chemin_fichier)

File: question1.php
<?php
function extraireSommeEtalonnage($cheminFichier) {
    $sommeEtalonnage = 0; // Initialiser la somme

    // Vérifier si le fichier existe
    if (!file_exists($cheminFichier)) {
        die("Erreur : Le fichier spécifié n'existe pas.\n");
    }

    // Lire le fichier ligne par ligne
    $fichier = fopen($cheminFichier, "r");
    $index = 1; // Compteur de ligne
    while (($ligne = fgets($fichier)) !== false) {
        $ligne = trim($ligne); // Supprimer les espaces inutiles
        $premierChiffre = '';
        $dernierChiffre = '';

        // Trouver le premier chiffre
        for ($i = 0; $i < strlen($ligne); $i++) {
            if (ctype_digit($ligne[$i])) {
                $premierChiffre = $ligne[$i];
                break;
            }
        }

        // Trouver le dernier chiffre
        for ($i = strlen($ligne) - 1; $i >= 0; $i--) {
            if (ctype_digit($ligne[$i])) {
                $dernierChiffre = $ligne[$i];
                break;
            }
        }

        // Former le nombre et l'ajouter à la somme
        if ($premierChiffre !== '' || $dernierChiffre !== '') {
            $nombreForme = (int) ($premierChiffre . $dernierChiffre);
            echo "Ligne $index: $premierChiffre$dernierChiffre\n"; // Afficher la concaténation
            $sommeEtalonnage += $nombreForme;
        } else {
            echo "Ligne $index: Aucun chiffre trouvé\n"; // Afficher s'il n'y a pas de chiffre
        }

        $index++; // Incrémenter le compteur de ligne
    }
    fclose($fichier);

    // Afficher la somme totale
    echo "\nLa somme des valeurs d'étalonnage est : $sommeEtalonnage\n";
}

// Définir le chemin du fichier
$cheminFichier = "C:/Users/arijh/Desktop/Test_Arij_Hamraoui/document.txt";

// Appeler la fonction pour calculer et afficher la somme
extraireSommeEtalonnage($cheminFichier);
?>
